aws-cpp-sdk-s3control/source/S3ControlClient.cpp

```diff
@@ -230,16 +230,15 @@ CreateAccessPointOutcome S3ControlClient::CreateAccessPoint(const CreateAccessPo
     AWS_LOGSTREAM_ERROR("CreateAccessPoint", "Required field: Name, is not set");
     return CreateAccessPointOutcome(Aws::Client::AWSError<S3ControlErrors>(S3ControlErrors::MISSING_PARAMETER, "MISSING_PARAMETER", "Missing required field [Name]", false));
   }
-  if (m_clientConfiguration.enableHostPrefixInjection)
```

Comment:
enableHostPrefixInjection is mandatory for Operations with endpoint trait, there is no reason in conditionally injecting it.